src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java

```diff
@@ -54,8 +53,10 @@ void dectectOnchainCompletedWhenOffchainCompleting(List<TaskStatus> dectectWhenO
                 }
 
                 boolean isReplicateStatusCompleting = lastRelevantStatus.get().equals(offchainCompleting);
+                boolean isStatusTrueOnchain = iexecHubService.isStatusTrueOnChain(task.getChainTaskId(),
+                        replicate.getWalletAddress(), onchainCompleted);
 
-                if (isReplicateStatusCompleting && iexecHubService.isStatusTrueOnChain(task.getChainTaskId(), replicate.getWalletAddress(), onchainCompleted)) {
+                if (isReplicateStatusCompleting && isStatusTrueOnchain) {
```

Comment:
I think `isReplicateStatusCompleting && iexecHubService.isStatusTrueOnChain(..)` is better since it avoids calling constantly a blockchain node

Reply:
"Bien vu" thanks!